File: frontend/src/types/catalog.ts
export interface Product {
  id: number;
  name: string;
  sku: string;
  category: string;
  unit: string;
  available_qty: number;
  is_stop: boolean;
  price?: number;
  image?: string;
  description?: string;
}

export interface CategoryGroup {
  category: string;
  items_count: number;
  items: Product[];
}

export interface CatalogResponse {
  total_categories: number;
  total_products: number;
  categories: CategoryGroup[];
}

export interface CoffeeBar {
  id: number;
  name: string;
  address?: string;
  telegram_chat_id?: number;
  is_active: boolean;
}

File: app/schemas/catalog.py
from typing import Dict, List
from pydantic import BaseModel, ConfigDict, Field


class CatalogProductItem(BaseModel):
    id: int
    name: str
    sku: str
    category: str
    unit: str
    available_qty: float = Field(..., description="Available quantity in warehouse (real_qty - reserved_qty)")
    is_stop: bool = Field(..., description="True if available_qty <= 0")

    model_config = ConfigDict(from_attributes=True)


class CatalogCategoryGroup(BaseModel):
    category: str
    items_count: int
    items: List[CatalogProductItem]


class CatalogResponse(BaseModel):
    total_categories: int
    total_products: int
    categories: List[CatalogCategoryGroup]
